docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -181,10 +181,19 @@ the `controlPlaneMachineType` variable accordingly.
 
 **Notes**: Same notes as in Story 4 apply.
 
+#### Story 8 - Easy UX for MachineHealthChecks
+As a cluster operator I want a simple way to define checks to manage the health of the machines in my cluster. 
+
+Instead of defining MachineHealthChecks each time a Cluster is created, there should be a mechanism for creating the same type of health check for each Cluster stamped by a ClusterClass.
+
+**Examples**: 
+- A cluster operator would like to create a single MachineHealthCheck that applies across each MachineDeployment and ControlPlane created for Clusters stamped by a ClusterClass.
```

Comment:
In most cases, we should try to separate the control-plane MHC from Machine Deployment ones, they might also have different defaults. For example, each control plane node might take longer to come up than worker nodes.

Reply:
To start with I think it's simpler to reconcile all defaults to the same - with the current webhook defaults providing most of the values. Later if we have a strong idea of how the defaults should differ between control planes and MHCs we can have multiple defaults.

Reply:
The control plane nodes are usually always different than the others, for example a control plane might take a lot longer to join that a worker node. We should probably avoid an overall set of parameters that's shared for all of the nodes, including control plane ones; from a user perspective, I'd expect that a healthCheck section under Control Plane to only apply to control plane nodes rather, and one under workers to apply to all workers instead.

Thoughts?

cc @CecileRobertMichon @fabriziopandini

Reply:
+1 to rephrase this; if fact we are planning to create 1 MHC for the CP and 1 for each MachineDeployment (vs having a single MachineHealthCheck)

Reply:
Yeah this examples section is out of sync. I've removed it completely. It was added to show how that use case - global MHC definition - would work, but it's no longer in the proposal.